compiler/ras/ILValidationStrategies.hpp

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2017 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+
+#ifndef ILVALIDATIONSTRATEGIES_HPP
+#define ILVALIDATIONSTRATEGIES_HPP
+
+#include <stdint.h>                     // for uint16_t
+
+namespace OMR {
+
+enum ILValidationRule
+   {
+   soundnessRule,
+   validateChildCount,
+   validateChildTypes,
+   validateLivenessBoundaries,
+   validateNodeRefCountWithinBlock,
+   validate_ireturnReturnType,
+   /**
+    * NOTE: Add `id`s for any new ILValidationContext here!
+    *       This needs to match the implementation of said *ILValidationRule.
+    *       Please see ILValidationRules.cpp for relevant examples.
+    */
+
+   /* Used to mark the end of a ILValidationStrategy */
+   endRules
+   };
+
+struct ILValidationStrategy
+   {
+   OMR::ILValidationRule        id;
+   /**
+    * TODO: Eventually ILValidationStrategy should be implemented
```

Comment:
Removed in the latest change list.

Since the said `TODO` does not pre-exist this PR, I will wait for this to go through before opening up an Issue. (If you're alright with that).